### PR TITLE
feat(sdk-coin-oas): add oas sdk skeleton

### PR DESCRIPTION
**Pull request by adarsh312**
Ticket: WIN-3696

### CODE REVIEW

File: modules/sdk-coin-oas/src/oas.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  protected getTransactionBuilder(): EthLikeTransactionBuilder {
+    return new TransactionBuilder(coins.get(this.getBaseChain()));
+  }
+
+  /** @inheritDoc */
+  allowsAccountConsolidations(): boolean {
+    return true;
+  }
```
Comment (gianchandania) [inline modules/sdk-coin-oas/src/oas.ts:29]: This is not needed as it is having forwarders

File: modules/sdk-coin-oas/src/oas.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  /** @inheritDoc */
+  getMPCAlgorithm(): MPCAlgorithm {
+    return 'ecdsa';
+  }
+
+  /**
+   * Make a query to Berachain explorer for information such as balance, token balance, solidity calls
```
Comment (gianchandania) [inline modules/sdk-coin-oas/src/oas.ts:42]: remove berachain

File: modules/sdk-coin-oas/src/oas.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+   * @param {Object} query key-value pairs of parameters to append after /api
+   * @returns {Promise<Object>} response Berachain explorer
```
Comment (gianchandania) [inline modules/sdk-coin-oas/src/oas.ts:44]: same as above

File: modules/sdk-coin-oas/test/integration/index.ts
Comment (gianchandania) [inline modules/sdk-coin-oas/test/integration/index.ts:1]: Seems like this empty file is committed by mistake

File: modules/sdk-coin-oas/test/unit/oas.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    it('should return the right info for oas', function () {
+      const oas = bitgo.coin('oas');
+
+      oas.should.be.an.instanceof(Oas);
+      oas.getChain().should.equal('oas');
+      oas.getFamily().should.equal('oas');
+      oas.getFullName().should.equal('oaschain');
+      oas.getBaseFactor().should.equal(1e18);
+      oas.supportsTss().should.equal(true);
+      oas.allowsAccountConsolidations().should.equal(true);
```
Comment (gianchandania) [inline modules/sdk-coin-oas/test/unit/oas.ts:27]: this should be false

File: modules/sdk-coin-oas/test/resources/oas.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import { KeyPair } from '@bitgo/abstract-eth';
+
+export const accounts = {
```
Comment (gianchandania) [inline modules/sdk-coin-oas/test/resources/oas.ts:3]: Please remove the ones that are not used

File: modules/sdk-coin-oas/test/unit/oas.ts (hunk 4 above)
Comment (adarsh312) [reply modules/sdk-coin-oas/test/unit/oas.ts:27]: do we support consolidation ?

File: modules/sdk-coin-oas/src/oas.ts (hunk 1 above)
Comment (gianchandania) [reply modules/sdk-coin-oas/src/oas.ts:29]: no need to add this method only as by default it is false in the BaseCoin, can remove in a follow up

**Review by gianchandania — DISMISSED** [review]

File: modules/sdk-coin-oas/test/fixtures/ecdsa.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
+export const keyShares = {
```
Comment (zahin-mohammad) [inline modules/sdk-coin-oas/test/fixtures/ecdsa.ts:1]: This looks like a GG18 keyshare. Can we convert this test to use DKLS instead?

**Review by zahin-mohammad — DISMISSED** [review]
Update code owners please.

**Review by gianchandania — APPROVED** [review]

**Review by therealdwright — APPROVED** [review]
CODEOWNERS LGTM.

**Pull request merged**